src/eth.rs

```diff
@@ -1032,6 +1032,24 @@ impl<'d, T> EthDriver<'d, T> {
         Ok(())
     }
 
+    pub fn set_promiscuous(&self, state: bool) -> Result<(), EspError> {
```

Comment:
1. Put some documentation above the method. 
2. Do not take &self but &mut self.